[blog] Post survey results 2020 (#21555)
mui-org/material-ui: docs/pages/blog/2020-developer-survey-results.md

Marija Najdova, Olivier Tassinari, Matt Brookes. June 24, 2019.

Continuing the tradition from last year, we launched a developer survey few months ago, to which we received 1488 contributions. This is twice as many as last year, so we thank you all for the participation!
The survey is now closed and here we will give a detailed summary of the results.

Like last year, the survey was again broken into three sections: "Introduction", "About you" and "Your product".

## Introduction

In this section we mostly wanted to hear what developers think is going well and that we should keep doing, and which are the areas where we need to improve the library to make it even better.

### 1. How would you feel if you could no longer use Material-UI?

<img src="/static/blog/2020-survey/1.png" style="width: 552px; margin-top: 32px; margin-bottom: 32px;" alt="Pie chart: 73.3% Very disappointed, 21.3% somewhat disappointed, 5.4% not disappointed." />

Similar to last year, over 94% of the respondents would be disappointed if they can no longer use Material-UI, which is very encouraging. We will keep working hard to hopefully move more of you into the "very disappointed" category!

> The number who would not be disappointed has moved down from 6.5% to 5.4%, which is a 17% improvement! We'd love to understand more about those who use Material-UI, but would happily use other solutions, so a follow up question might be needed next year.

Reviewer: Perhaps should acknowledge that this is a bit cheeky. 🙂 
```suggestion
The number who would not be disappointed has moved down from 6.5% to 5.4%, which is technically a 17% improvement! 🙂 We'd love to understand more about those who use Material-UI, but would happily use other solutions, so a follow up question might be needed next year.
```

---
description: 2020 Material-UI Developer Survey results
---

# 2020 Material-UI Developer Survey results

While we are currently working on the upcoming release of Material-UI v5, we need to prioritize our
roadmap for the coming year. To refine our focus, we launched a developer survey few months back,

> to which we received 1488 contributions. This year we had twice as more contributions than last year,

Reviewer: ```suggestion
to which we received 1488 contributions. This year we had twice as many contributions as last year,
```

so thank you for your participation!
The survey is now closed and this is a summary of the results.

The survey was broken into three sections, "Introduction", "About you" and "Your product".

## Introduction

Here we asked for the good, the bad and the ugly. Thankfully, it's mostly good, but we can focus

> on the things that will make it even better.

Reviewer: Could we have some new commentary throughout, rather than copying last years? I don't know how many developers will notice (or care), but it'll get a negative score as duplicate content from an SEO perspective.

Author: Updated the text, please take a look again.

### 2. How likely is it that you would recommend Material-UI to a friend or colleague?

<img src="/static/blog/2020-survey/2a.png" style="display: block; margin: 0 auto;" alt="Bar chart: 0.20% - 1, 0% - 2, 0.20% - 3, 0.20% - 4, 1.02% - 5, 1.97% - 6, 8.71% - 7, 22.52% - 8, 20.88% - 9, 44.29% - 10" />
<br />
<img src="/static/blog/2020-survey/2b.png" style="display: block; margin: 0 auto;" alt="Pie chart: 73.3% Very disappointed, 21.3% somewhat disappointed, 5.4% not disappointed." />

Developers gave Material-UI a [Net Promoter Score](https://en.wikipedia.org/wiki/Net_Promoter)
(promoters less detractors) of 61.54%. Given the NPS range of -100 to +100, a “positive” score or NPS
above 0 is considered “good”, +50 is “Excellent”, and above 70 is considered “world class.”

### 3. Who do you think would most benefit from Material-UI?

This word cloud was generated with [wordclouds.com](https://wordclouds.com).

<img src="/static/blog/2020-survey/3.jpg" style="display: block; margin: 0 auto;" alt="Word cloud of who would benefit most" />

Developers, developers, developers! (And "people" and "teams" in general, apparently! ) This is understandable, given the

> [job role demographic](#7-which-of-the-following-best-describes-your-current-job-role)

Reviewer 2: ```suggestion
[job role demographic](#8-which-of-the-following-best-describes-your-current-job-role)
```

of the majority of respondents. We'll continue to work on the developer experience, making it
easier to get started with, use, and customize Material-UI.

### 4. What is the main benefit you receive from Material-UI?

<img src="/static/blog/2020-survey/4.jpg" style="display: block; margin: 0 auto;" alt="Word cloud of the main benefit of Material-UI" />

The responses variously commented on being able to focus on the business logic, faster development,
good looking components, ease of use, design consistency, good documentation.
We'll make sure that we continue to do more of the things that benefit you the most.

### 5. How can we improve Material-UI for you?

We have grouped the answers into different categories.
The prefix corresponds to the number of times the concern was mentioned.
It’s sorted descending, with the most important concerns first:

> <style>th { text-align: left; background-color: #f2f2f2; }</style>

Reviewer 3: This background colour makes text nearly impossible to read in dark mode.

Reviewer: We also need dark mode images.

Reviewer 2: What about we disable the dark mode for the blog?

  <tr><th>7</th><th> mobile</th><tr>
  <tr><th>7</th><th> less breaking changes</th><tr>
  <tr><th>7</th><th> designers</th><tr>
  <tr><td>2</td><td>designers - bridge design tools and code</td><tr>
  <tr><td>2</td><td>designers - Adobe XD material</td><tr>
  <tr><td>2</td><td>designers - Figma material</td><tr>
  <tr><td>1</td><td>designers - ?</td><tr>
  <tr><th>6</th><th> free vs paid balance</th><tr>
  <tr><th>6</th><th> tree view - improve</th><tr>
  <tr><th>6</th><th> less abstracted components</th><tr>
  <tr><th>5</th><th> more abstracted components</th><tr>
  <tr><th>4</th><th> accessibility</th><tr>
  <tr><td>1</td><td>accessibility - ?</td><tr>
  <tr><td>1</td><td>accessibility - auto id</td><tr>
  <tr><td>1</td><td>accessibility - full audit</td><tr>
  <tr><td>1</td><td>accessibility - more examples</td><tr>
  <tr><th>4</th><th> system</th><tr>
  <tr><td>1</td><td>system - CSS grid</td><tr>
  <tr><td>1</td><td>system - docs</td><tr>
  <tr><td>1</td><td>system - performance</td><tr>
  <tr><td>1</td><td>system - rework breakpoints</td><tr>
  <tr><th>3</th><th> components consistency</th><tr>
  <tr><th>3</th><th> grid - improve</th><tr>
</table>

> ### 6. What're your most important criteria for choosing a UI library?

Reviewer: ```suggestion
### 6. What are your key criteria when choosing a UI library?
```

("What're" isn't a valid contraction)

---
description: 2020 Material-UI Developer Survey results
---

# 2020 Material-UI Developer Survey results

> Marija Najdova, Olivier Tassinari, Matt Brookes. June 24, 2019.

Reviewer 3: 2019?

Continuing the tradition from last year, we launched a developer survey few months ago, to which we received 1488 contributions. This is twice as many as last year (734), so we thank you all for the participation!
The survey is now closed and here we will give a detailed summary of the results.

Like last year, the survey was again broken into three sections: ["Introduction"](#Introduction), ["About you"](#about-you) and ["Your product"](#your-product).

## Introduction

In this section we mostly wanted to hear what developers think is going well and that we should keep doing, and which are the areas where we need to improve the library to make it even better.

### 1. How would you feel if you could no longer use Material-UI?

<img src="/static/blog/2020-survey/1.png" style="width: 552px; margin-top: 32px; margin-bottom: 32px;" alt="Pie chart: 73.3% Very disappointed, 21.3% somewhat disappointed, 5.4% not disappointed." />

> Similar to last year, over 94% of the respondents would be disappointed if they can no longer use Material-UI, which is very encouraging. We will keep working hard to hopefully move more of you into the "very disappointed" category!

Reviewer 3: ```suggestion
Similar to last year, over 94% of the respondents would be disappointed if they could no longer use Material-UI, which is very encouraging. We will keep working hard to hopefully move more of you into the "very disappointed" category!
```

> The number who would not be disappointed has moved down from 6.5% to 5.4%, which is technically a 17% improvement! 🙂 We'd love to understand more about those who use Material-UI, but would happily use other solutions, so a follow-up question might be needed next year.

Reviewer 3: ```suggestion
The number of respondents who would not be disappointed has moved down from 6.5% to 5.4%, which is technically a 17% improvement! 🙂 We'd love to understand more about those who use Material-UI, but would happily use other solutions, so a follow-up question might be needed next year.
```

(promoters less detractors). This year it is again a pretty high number 61.54%! (As the values for NPS range between -100 and +100, a “positive” score is considered “good”, greater than 50 is “excellent”, and above 70 is considered “world class”).

### 3. Who do you think would most benefit from Material-UI?

<img src="/static/blog/2020-survey/3.jpg" style="width: 512px; margin-top: 32px; margin-bottom: 16px;" alt="Word cloud of who would benefit most" />

<p class="blog-description">This word cloud was generated with <a href="https://wordclouds.com">wordclouds.com</a>.</p>

Developers are again at the center of our universe 🙂 (And "people" and "teams" in general, of course!) This is understandable, given the [job role demographic](#8-which-of-the-following-best-describes-your-current-job-role)
of the majority of respondents. We will push hard on making the experience for you even better over the next year.

### 4. What is the main benefit you receive from Material-UI?

<img src="/static/blog/2020-survey/4.jpg" style="width: 512px; margin-top: 32px; margin-bottom: 32px; " alt="Word cloud of the main benefit of Material-UI" />

> The responses to this question are a very clear indicator to us about what we need to continue to do more of. Some of the most common points were the range of components, ease of use, documentation quality, as well as the design. We will of course continue to work on all of these.

Reviewer 3: ```suggestion
The responses to this question are a very clear indicator to us about what we need to continue to do more of. Some of the most common points were: the range of components, ease of use, documentation quality, as well as the design. We will, of course, continue to work on all of these.
```

### 5. How can we improve Material-UI for you?

As the answers to these questions were pretty different, we grouped them into different categories and counted the different number of times the concern was mentioned. You can see all of them sorted in descending order:

<style>th { text-align: left; background-color: #f2f2f2; }</style>

<table>

>   <tr><th>306</th><th> more components</th><tr>

Author: There shouldn't be any spaces, good catch will fix those 👍

  <tr><th>7</th><th> designers</th><tr>
  <tr><td>2</td><td>designers - bridge design tools and code</td><tr>
  <tr><td>2</td><td>designers - Adobe XD material</td><tr>
  <tr><td>2</td><td>designers - Figma material</td><tr>
  <tr><td>1</td><td>designers - ?</td><tr>
  <tr><th>6</th><th> free vs paid balance</th><tr>
  <tr><th>6</th><th> tree view - improve</th><tr>
  <tr><th>6</th><th> less abstracted components</th><tr>
  <tr><th>5</th><th> more abstracted components</th><tr>
  <tr><th>4</th><th> accessibility</th><tr>
  <tr><td>1</td><td>accessibility - ?</td><tr>
  <tr><td>1</td><td>accessibility - auto id</td><tr>
  <tr><td>1</td><td>accessibility - full audit</td><tr>
  <tr><td>1</td><td>accessibility - more examples</td><tr>
  <tr><th>4</th><th> system</th><tr>
  <tr><td>1</td><td>system - CSS grid</td><tr>
  <tr><td>1</td><td>system - docs</td><tr>
  <tr><td>1</td><td>system - performance</td><tr>
  <tr><td>1</td><td>system - rework breakpoints</td><tr>
  <tr><th>3</th><th> components consistency</th><tr>
  <tr><th>3</th><th> grid - improve</th><tr>
</table>

### Comparison with last year

There are a couple of noticeable differences compared to last year.

> Some can be explained by our work, others by an evolution of the ecosystem.

Reviewer 3: ```suggestion
Some can be explained by our work, others by the evolution of the ecosystem.
```

Each item is prefixed by the multiplication factor of the pain point for 2020 relative to 2019.

Decreasing pain:
- x0: Slider. No requests. The requirements are mostly met, especially with the introduction of the range feature.

> - x0.1: Strict mode. We did a lot for it this year. However, since Create React App has made this mode a default, we have seen a lot more requests for it.

Reviewer 3: ```suggestion
- x0.1: Strict mode. We fixed a lot of strict mode compatibility issues this year. However, since Create React App has made this mode a default, we have seen a lot more requests for it.
```

- x0.1: Autocomplete. We added a new component, and have resolved a large number of issues opened since. This will be moved from lab to the core in v5.
- x0.2: Fewer breaking changes. Only releasing minor versions under v4 for over a year helps a lot. However, we still need to be careful with CSS changes.
- x0.3: Accessibility. We did a lot for it this year. One of our best leverages has been GitHub's issue opened by a11y experts, often coming from large companies using Material-UI at scale.
- x0.4: Material design. We didn't do much for it this year, at least not as much as we could have. Maybe the reduction is because fewer people care? It seems that we start to resonate more with developers building custom design systems.

> - x0.4: TypeScript. Our effort is paying off with the migration of all the demos to TypeScript and the migration of all the props to IntelliSense.

Reviewer 3: ```suggestion
- x0.4: TypeScript. The continued migration of all the demos to TypeScript and of all the props to IntelliSense is paying off.
```

- x0.5: Date picker. We did a lot for it this year. We probably still need the range feature, and to move it into the main repository (docs migration, etc.) for consistency.
- x0.7: Performance. We didn't do much this year, so perhaps developers are leveraging React more effectively with virtualization, update pruning, etc?

Growing pains:
- ∞: Forms is a new item. It seems that we should at least work more closely with react-hook-form, formik, and react-final-form.
- ∞: Charts is a new item. Material Design even has a page dedicated to [date visualization](https://material.io/design/communication/data-visualization.html).
- x5: Custom themes.
- x5: Simpler customization. We have improved customizability this year by introducing global class names and reducing the CSS specificity of some selectors. However, it seems that we are now tapping into a new audience. We need to do better.
- x1.5: Animations.
- x1.2: React native. We still have no plans for it. The [market is too small](https://npm-stat.com/charts.html?package=react-dom,react-native) to make it sustainable with our model.

> - x1.1: More components, the more we give, the more developers ask for. We will fully solve this with the enterprise version. The first early access will land this year.

Reviewer 2: ```suggestion
- x1.1: More components, the more we give, the more developer asks for. We will try to better solve this with the enterprise version. Both because it's the best model we have found that can sustain the developments of data-rich advanced components and because it allows us to invest back resources into the MIT side. The first early access will land this year.
```

The trend seems to be similar to the one from the last year's survey, so we will push with better support for styled components.

### 20. Has your organization ever paid for UI components?

<img src="/static/blog/2020-survey/20.png" style="width: 553px; margin-top: 32px;" alt="Pie chart: 89.90%	No, 10.10% Yes" />

### 21. What type system are you using?

<img src="/static/blog/2020-survey/21.png" style="width: 614px; margin-top: 32px;" alt="Bar chart: 54.08%	None
30.87%	TypeScript 3.8, 7.31%	TypeScript 3.7, 1.90%	Flow, 1.55%	TypeScript 3.6, 0.98%	TypeScript 3.5, 3.31%	Other" />

Almost half of the community is using TypeScript. Next year, it might even become more popular than JavaScript.

## Conclusion

This data is **incredibly valuable** for our team.
Thank you again for participating!
We want to work on the problems that resonate the most with our users.
[It's clear](#5-how-can-we-improve-material-ui-for-you) that we should:

1. Provide more flexibility on the components, unstyled components (pure hooks?).
1. Make the customization easier and implement custom themes with Material-UI. Maybe provide a theme builder.

> 1. Irone the details out by providing a second theme, updates to match Material Design, providing more simple components and features (e.g. dropzone, carousel), providing a better DX (they are good ideas to apply from other UI libraries to Material-UI v5).

Reviewer: ```suggestion
1. Provide a second theme, update the current components to better match Material Design, provide more simple components and features (e.g. dropzone, carousel) as well as provide a better DX (there are good ideas from other UI libraries to apply to Material-UI v5).
```

> 1. Provide a second theme, do updates on the components to match Material Design, provide more simple components and features (e.g. dropzone, carousel) as well as provide a better DX (they are good ideas to apply from other UI libraries to Material-UI v5).

Reviewer 2: ```suggestion
1. Provide a second theme, update the current components to better match Material Design, provide more simple components and features (e.g. dropzone, carousel) as well as provide a better DX (there are good ideas from other UI libraries to apply to Material-UI v5).
```

> 1. Irone the details out by providing a second theme, updates to match Material Design, providing more simple components and features (e.g. dropzone, carousel), providing a better DX (they are good ideas to apply from other UI libraries to Material-UI v5).

Author: I think this was added by @oliviertassinari , but we decided to remove the comments for enlarging the users database, so the meaning may be lost now.. I believe we can restructure it to
```suggestion
1. Provide a second theme, do updates on the components to match Material Design, provide more simple components and features (e.g. dropzone, carousel) as well as provide a better DX (they are good ideas to apply from other UI libraries to Material-UI v5).
```

> 1. Improve on the paid advanced version of the components (e.g. complex data grid, data range picker, tree view drag & drop, virtualization, etc).

Reviewer 3: ```suggestion
1. Improve upon the paid advanced versions of the components (e.g. complex data grid, date range picker, tree view drag & drop, virtualization, etc).
```

# 2020 Material-UI Developer Survey results

Marija Najdova, Olivier Tassinari, Matt Brookes. June 24, 2019.

> Continuing the tradition from last year, we launched a developer survey few months ago, to which we received 1488 contrivutions. This is twice as many as last year, so we thank you all for the participation!

Reviewer: ```suggestion
Continuing the tradition from last year, we launched a developer survey few months ago, to which we received 1488 contributions. This is twice as many as last year, so we thank you all for the participation!
```

The survey is now closed and here we will give a detailed summary of the results.

> The survey again was broken into three sections, "Introduction", "About you" and "Your product".

Reviewer: ```suggestion
Like last year, the survey was again broken into three sections: "Introduction", "About you" and "Your product".
```

## Introduction

> In this section we mostly wanted to hear what do developers think is going good and we should keep doing, and which are the areas where we need to improve the library to make it even better.

Reviewer: ```suggestion
In this section we mostly wanted to hear what developers think is going well and that we should keep doing, and which are the areas where we need to improve the library to make it even better.
```

### 1. How would you feel if you could no longer use Material-UI?

<img src="/static/blog/2020-survey/1.png" style="width: 552px; margin-top: 32px; margin-bottom: 24px;" alt="Pie chart: 73.3% Very disappointed, 21.3% somewhat disappointed, 5.4% not disappointed." />

> Over 94% of the respondents would be disappointed of they can no longer use Material-UI, which is really a nice percentage. We will keep working hard and hopefully move more of you into the "very disappointed" category soon!

Reviewer: ```suggestion
Similarto last year, over 94% of the respondents would be disappointed if they can no longer use Material-UI, which is very encouraging. We will keep working hard to hopefully move more of you into the "very disappointed" category!

The number who would not be disappointed has moved down from 6.5% to 5.4%, which is a 17% improvement! We'd love to understand more about those who use Material-UI, but would happily use other solutions, so a follow up question might be needed next year.
```

### 2. How likely is it that you would recommend Material-UI to a friend or colleague?

<img src="/static/blog/2020-survey/2a.png" style="width: 550px;" alt="Bar chart: 0.20% - 1, 0% - 2, 0.20% - 3, 0.20% - 4, 1.02% - 5, 1.97% - 6, 8.71% - 7, 22.52% - 8, 20.88% - 9, 44.29% - 10" />
<br />
<img src="/static/blog/2020-survey/2b.png" style="width: 550px;" alt="Pie chart: 73.3% Very disappointed, 21.3% somewhat disappointed, 5.4% not disappointed." />

As previous year, we again calculated the [Net Promoter Score](https://en.wikipedia.org/wiki/Net_Promoter)

> (promoters less detractors). This year it is again a pretty high number 61.54%! (As the values for the NPS range are between -100 and +100, a “positive” score or NPS is considered “good”, +50 is “Excellent”, and above 70 is considered “world class”).

Reviewer: ```suggestion
(promoters less detractors). This year it is again a pretty high number 61.54%! (As the values for NPS range between -100 and +100, a “positive” score is considered “good”, greater than 50 is “excellent”, and above 70 is considered “world class”).
```

### 3. Who do you think would most benefit from Material-UI?

This word cloud was generated with [wordclouds.com](https://wordclouds.com).

<img src="/static/blog/2020-survey/3.jpg" style="width: 550px;" alt="Word cloud of who would benefit most" />

> Developers are again in the center of our universe :) (And "people" and "teams" in general, of course!) This is understandable, given the [job role demographic](#8-which-of-the-following-best-describes-your-current-job-role)

Reviewer: ```suggestion
Developers are again at the center of our universe 🙂 (And "people" and "teams" in general, of course!) This is understandable, given the [job role demographic](#8-which-of-the-following-best-describes-your-current-job-role)
```

of the majority of respondents. We will push hard on making the experience for you even better over the next year.

### 4. What is the main benefit you receive from Material-UI?

<img src="/static/blog/2020-survey/4.jpg" style="width: 550px;" alt="Word cloud of the main benefit of Material-UI" />

> These responses on this question are very clear indicator to us about what we need to continue to do more. Some of the most repetating points were the palette of components, the ease of use, the documentation, as well as the design. We will for sure continue to work on all of these.

Reviewer: ```suggestion
The responses to this question are a very clear indicator to us about what we need to continue to do more of. Some of the most common points were the range of components, ease of use, documentation quality, as well as the design. We will of course continue to work on all of these.
```

### 5. How can we improve Material-UI for you?

> As the answers to this questions were pretty different, we grouped them into different categories and counted the different number of times the concern was mention. You can see all of them sorted by descending order, where the most important concerns are listed first:

Reviewer: ```suggestion
As the answers to this questions were pretty different, we grouped them into different categories and counted the different number of times the concern was mentioned. You can see all of them sorted in descending order:
```

  <tr><th>3</th><th> grid - improve</th><tr>
</table>

### Comparison with last year

There are a couple of noticeable differences compared to last year.
Some can be explained by our work, others by an evolution of the ecosystem.

> Each item is prefixed by the multiplicator factor of pain point between 2019 and 2020.

Reviewer: ```suggestion
Each item is prefixed by the multiplication factor of the pain point for 2020 relative to 2019.
```

Decreasing pain:

> - x0: Slider. No requests, the problem is almost solved since the introduction of the range feature.

Reviewer: ```suggestion
- x0: Slider. No requests. The requirements are mostly met, especially with the introduction of the range feature.
```

- x0.1: Strict mode. We did a lot for it this year. However, since Create React App has made this mode a default, we have seen a lot more requests for it.

> - x0.1: Autocomplete. We did a lot for it this year. The component will be moved to the core in v5.

Reviewer: ```suggestion
- x0.1: Autocomplete. We added a new component, and have resolved a large number of issues opened since. This will be moved from lab to the core in v5.
```

- x0.2: Fewer breaking changes. Only releasing minor versions under v4 for over a year helps a lot. However, we still need to be careful with CSS changes.

> - x0.3: Accessibility. We did a lot for it this year. One of our best leverage has been GitHub's issue opened by a11y expert, often coming from large companies using Material-UI at scale.

Reviewer: ```suggestion
- x0.3: Accessibility. We did a lot for it this year. One of our best leverages has been GitHub's issue opened by a11y experts, often coming from large companies using Material-UI at scale.
```

> - x0.4: Material design. We didn't do much for it this year, at least not as much as we could have. Maybe fewer people care? It seems that we resonate better with developers going for custom design systems.

Reviewer: ```suggestion
- x0.4: Material design. We didn't do much for it this year, at least not as much as we could have. Maybe the reduction is because fewer people care? It seems that we start to resonate more with developers building custom design systems.
```

- x0.4: TypeScript. Our effort is paying off with the migration of all the demos to TypeScript and the migration of all the props to IntelliSense.

> - x0.5: Date picker. We did a lot for it this year. We probably still need the range feature and the integration in the main repository (docs migration, etc.) for consistency to get is much lower, to the level of the "Slider".

Reviewer: ```suggestion
- x0.5: Date picker. We did a lot for it this year. We probably still need the range feature, and to move it into the main repository (docs migration, etc.) for consistency to get is much lower, to the level of the "Slider".
```

> for consistency to get is much lower, to the level of the "Slider"

I don't understand this bit...

> - x0.7: Performance. We didn't do much this year, maybe more developers are used to leverage React effectively? Virtualization updates pruning, etc.

Reviewer: ```suggestion
- x0.7: Performance. We didn't do much this year, so perhaps developers are leveraging React more effectively with virtualization, update pruning, etc?
```

> Growing pain:

Reviewer: ```suggestion
Growing pains:
```

> - ∞: Form is a new item. It seems that we should at least work more closely with react-hook-form, formik, and react-final-form.

Reviewer: ```suggestion
- ∞: Forms is a new item. It seems that we should at least work more closely with react-hook-form, formik, and react-final-form.
```

> - ∞: Charts is a new item. Material Design even has a page dedicated to [date visualization](https://material.io/design/communication/data-visualization.html) since recently.

Reviewer: > since recently

I'm sure it's at least a year old...

Reviewer 2: You are right, I can trace it back to July 2019.
```suggestion
- ∞: Charts is a new item. Material Design even has a page dedicated to [date visualization](https://material.io/design/communication/data-visualization.html).
```

- x5: Custom themes.

> - x5: Simpler customization. We have improved customizability this year by introducing global class names and reducing the CSS specificity of some selectors. However, it seems that we are now tapping into a new audience. We need to do better.

Reviewer: One problem is perhaps that we almost have too many customisation options. It can perhaps be hard to know where to start.

Reviewer 2: Very possible, a non-negligeabe amount of developers ask for documentation more welcoming to new users, tutorials, and a more opinionated answer (one way).

- x1.5: Animations.
- x1.2: React native. We still have no plans for it. The [market is too small](https://npm-stat.com/charts.html?package=react-dom,react-native) to make it sustainable with our model.

> - x1.1: More components, the more we give, the more developer asks for, we will fully solve this with the enterprise version. The first early access will land this year.

Reviewer: > we will fully solve this

That isn't true. There will always be a long tail of niche components that we don't provide.

> with the enterprise version

And that will be the opposite - a limited set of advanced components.

Reviewer 2: ```suggestion
- x1.1: More components, the more we give, the more developer asks for. We will try to better solve this with the enterprise version. Both because it's the only model we have found that can sustain the developments of data-rich advanced components and because it allows us to invest back resources into the MIT side. The first early access will land this year.
```

## About you

### 7. How did you hear about Material-UI?

<img src="/static/blog/2020-survey/7.png" style="width: 550px;" alt="Bar chart: 50.65%	Search, 26.18%	Word of mouth, 10.76%	Social, 5.10%	Blog, 7.31%	Other." />

### 8. Which of the following best describes your current job role?

<img src="/static/blog/2020-survey/8.png" style="width: 550px;" alt="Bar chart: 52.18%	Full-stack developer, 27.11%	Frontend developer, 11.65%	Entrepreneur (I do it all), 3.47%	Beginner learning web development, 1.23%	Backend Developer, 1.16%	UX Designer, 0.34%	Student, 0.20%	CTO, 2.66%	Other." />

This was expected :) 

### 9. How big is your organization?

<img  src="/static/blog/2020-survey/9.png" style="width: 550px;" alt="Bar chart: 12.30%	Hobby / side project, 13.40%	Self-employed, 35.60%	2-5 people, 16.10%	6-10, 10.30%	11-20, 5.10%	21-50, 2.50%	51-100, 4.60%	100+" />

> Seems like we are continiosly most popular with small to medium sized organisations, so we will keep working on the ease of use of the components, while at the same time, allowing designers to give the appearence of their organization into each one of them.

Reviewer: ```suggestion
It seems we are consistently popular with small to medium sized organisations, so we will keep working on the ease of use of the components, while at the same time, allowing designers to style them to match their organisations brand.
```

### 10. How long have you been developing with JavaScript

<img src="/static/blog/2020-survey/10.png" style="width: 550px;" alt="Bar chart: 3.60%	I'm just getting started!, 7.40%	6 months +, 20.80%	1 year +, 27.80%	3 years +, 24.30%	5 years +, 9.80%	10 years +, 4.10%	15 years +, 2.30%	20 years +" />

> We can see the normal distrubution of the developers experience here, where most of them have a "mid" level of experience (between 1 and 5 years).

Reviewer: ```suggestion
We can see the normal distribution of developer experience here, where most have moderate level of experience (between 1 and 5 years).
```

### 11. How long have you been developing with React?

<img src="/static/blog/2020-survey/11.png" style="width: 550px;" alt="Bar chart: 10.70%	I'm just getting started!, 15.60%	6 months +, 24.60%	1 year +, 21.40%	2 years +, 23.10%	3 years +, 4.60%	Bleeding edge baby! 5 years +" />

### 12. How long have you been developing with Material-UI?

<img src="/static/blog/2020-survey/12.png" style="width: 550px;" alt="Bar chart: 20.50%	I'm just getting started!, 24.90%	6 months +, 28%	1 year +, 17.40%	2 years +, 7.50%	3 years +, 1.70%	I'm a pioneer! 4 years +" />

### 13. What were you primarily using before Material-UI?

<img src="/static/blog/2020-survey/13.png" style="width: 550px;" alt="Bar chart: 47.08%	Bootstrap, 16.04%	Custom system, 13.68%	Started with Material-UI, 6.67%	Angular Material, 4.44%	Semantic-UI, 3.19%	Ant Design, 8.89%	Other" />

> Similar to last year, it seems that still most of the responders were previously using Bootstrap. We can see that also the custom solutions were replaced with Material-UI as well as some other frameworks.

Reviewer: ```suggestion
Similar to last year, it seems that most respondents were previously using Bootstrap. We can see also that custom solutions, as well as some other frameworks, were replaced with Material-UI.
```

### 14. How many Material-UI based projects have you built?

<img src="/static/blog/2020-survey/14.png" style="width: 550px;" alt="Bar chart: 11.50%	0 (I’m just getting started), 23%	1, 54.80%	2-5, 7.80%	6-10, 2.90%	10+" />

## Your product

### 15. What are you building?

<img src="/static/blog/2020-survey/15.png" style="width: 550px;" alt="Bar chart: 23.43%	Enterprise application, 20.31%	Dashboard admin app, 7.40%	A design system, 7.40%	A set of UI components, 7.37%	playing with tech, 7.34%	A prototype, 6.40%	Landing page, 5.41%	e-commerce site, 4.80%	CMS, 3.59%	Portfolio or resume, 1.88%	Blog, 4.69%	Other" />

> Enterprises and dashboards are at the top of the heap, and yes, we know that for many of you, the system you are building are internal, but if some of you have an system that you would like to share as part of [the showcase](/discover-more/showcase/), let us know by opening a PR. Also for those of you building UI components, we'd be happy to give you a shout out in the [related projects](/discover-more/related-projects/)

Reviewer: ```suggestion
Enterprises and dashboards are at the top of the heap, and yes, we know that for many of you, the systems you are building are internal, but if you have something that you would like to share as part of [the showcase](/discover-more/showcase/), let us know by opening a PR. Also for those of you building UI components, we'd be happy to give you a shout out in the [related projects](/discover-more/related-projects/)
```

<img  src="/static/blog/2020-survey/9.png" style="width: 589px; margin-top: 32px; margin-bottom: 8px;" alt="Bar chart: 12.30%	Hobby / side project, 13.40%	Self-employed, 35.60%	2-5 people, 16.10%	6-10, 10.30%	11-20, 5.10%	21-50, 2.50%	51-100, 4.60%	100+" />

> It seems we are consistently popular with small to medium-sized organizations, so we will keep working on the ease of use of the components, while at the same time, allowing designers to style them to match their organizations brand.

Reviewer: ```suggestion
It seems we are consistently popular with small to medium-sized organizations, so we will keep working on the ease of use of the components, while at the same time, allowing designers to style them to match their organizations' brand.
```

<img src="/static/blog/2020-survey/19.png" style="width: 728px; margin-top: 32px; margin-bottom: 24px;" alt="Pie chart: 53.84%	Material-UI styles (JSS), 20.41%	Styled components, 13.01%	Good plain CSS, 8.31%	CSS Modules, 1.96%	Emotion, 0.59%	scss, 0.59%	sass, 0.09%	less, 1.19%	Other" />

> The trend seems to be similar to the one from the last year's survey, so we will push with better support for styled components.

Reviewer: ```suggestion
The response seems to be similar to the one from the last year's survey, so we will push with better support for styled components.
```
?

### 20. Has your organization ever paid for UI components?

<img src="/static/blog/2020-survey/20.png" style="width: 553px; margin-top: 32px;" alt="Pie chart: 89.90%	No, 10.10% Yes" />

> 

Reviewer: ```suggestion
```

### 21. What type system are you using?

<img src="/static/blog/2020-survey/21.png" style="width: 614px; margin-top: 32px;" alt="Bar chart: 54.08%	None
30.87%	TypeScript 3.8, 7.31%	TypeScript 3.7, 1.90%	Flow, 1.55%	TypeScript 3.6, 0.98%	TypeScript 3.5, 3.31%	Other" />

> Almost half of the community is using TypeScript. Next year, it might even get more popular than JavaScript.

Reviewer: ```suggestion
Almost half of the community is using TypeScript. Next year, it might even become more popular than JavaScript.
```

## Conclusion

This data is **incredibly valuable** for our team.
Thank you again for participating!
We want to work on the problems that resonate the most with our users.
[It's clear](#5-how-can-we-improve-material-ui-for-you) that we should:

> 1. Increase the front-end user base by providing more flexibility on the components, unstyled components (pure hooks?).

Reviewer: ```suggestion
1. Increase the front-end user base by providing more flexibility on the components, unstyled components (pure hooks?).
```
Increasing the user base isn't something we should do to help the developer community.

Reviewer 2: I don't follow:

- We benefit from a network effect. Each new developer using Material-UI adds value to the existing users. Most developers are well aware of this effect. One example of this is StackOverflow.
- We don't increase the userbase by saying we want to. We have to solve real problems to do such, isn't that a good thing? More developers finding shoes to their feet?

Reviewer: If you read this from the perspective of a current or potential developer, not from that of the author, you'll hopefully see why this is the wrong motivation to attach these to. It's about DX, not growth. Growth is a secondary effect of improving the DX.

Reviewer 2: Agree for focusing on the pain points

Author: Done

> 1. Increase the designer user base by making it easier to customize and implement custom themes with Material-UI. Maybe provide a theme builder.

Reviewer: Same comment

> 1. Increase the full-stack user base by ironing the details out, providing a second theme, updates to match Material Design, providing more simple components and features (e.g. dropzone, carousel), providing a better DX (they are good ideas to apply from other UI libraries to Material-UI v5).

Reviewer: Ditto

> 1. Increase the enterprise user base by improving on the paid advanced version of the components (e.g. complex data grid, data range picker, tree view drag & drop, virtualization, etc).

Reviewer: Likewise
